text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+
+- New entities added by extension should not break existing code
+- Collector augmentation / enrichment (resource, e.g.) - Should be extensible and not hard-coded. We need a general algorithm not specific rulesets.
+
+### What identity should entities use (LID, UUID / GUID, or other)?
+
+One of the largest questions in the first entities' OTEP was how to identify an entity.  This was an attempt to unify the need for Navigational attributes with the notion that only identifying attributes of Entity would show up in Resource going forward. This restriction is no longer necessary in this proposal and we should reconsider how to model identity for an Entity.  
+
+This can be done in follow up design / OTEPs.
+
+### What happens if existing Resource translation in the collector remove resource attributes an Entity relies on?
+
+While we expect the collector to be the first component to start engaging with Entities in an architecture, this could lead to data model violations.  We have a few options to deal with this issue:
+
+- Consider this a bug and warn users not to do it.
+- Specify that missing attribute keys are acceptable for descriptive attribtues.
+- Specify that missing attribute keys denote that entities are unusable for that batch of telemetry, and treat the content as malformed.
+
+## Trade-offs and mitigations
+
+The design proposed here attempts to balance non-breaking (backwards and forwards compatible) changes with the need to improve problematic issues in the Specification.  Given the inability of most SDKs to implement the current Resource merge specification, breaking this should have little effect on actual users.  Instead, the proposed merge specification should allow implementations to match current behavior and expectation, while evolving for users who engage with the new model.
+
+## Prior art and alternatives
+
+Previously, we have a few unaccepted oteps, e.g. ([OTEP 208](https://github.com/open-telemetry/oteps/pull/208)).  Additionally, there are some alternatives that were considered in the Entities WG and rejected.
+
+Below is a brief discussion of some design decisions:
+
+- **Only associating one entity with a Resource.**  This was rejected, as too high a friction point in evolving semantic conventions and allowing independent systems to coordinate identity + entities within the OpenTelemetry ecosystem.  Eventually, this would force OpenTelemetry to model all possibly entities in the world and understand their interaction or otherwise prevent non-OpenTelemetry instrumentation from interacting with OpenTelemetry entities.
```

Comment:
Can you clarify how does this result in a friction in evolving semantic conventions? This seems orthogonal to me, I don't understand what does it have to do with the ability to record one associated entity.

Reply:
From experience [having a system like this](https://cloud.google.com/monitoring/api/resources) - it lead to everyone pushing for using the same entity (thereby trying to overload its definition) or refusing to define new ones / re-using the ones "close" to what they need, because of the difficulting in reporting against different entities.

E.g. having a Collector which defines the entity/resource lookup ONCE for all producers of telemetry is a use case that becomes difficult if that collector needs different entity types by source of data.  To the point folks just re-use the one it detect by default, negating the value of Entity as you need to check attributes for identity anyway.

Additionally, having new concepts be modeled AFTER the fact is hugely disruptive and breaking of the entire ecosystem to the point we just don't.  If you realize that there is a sub-concept in the model too late in the process, it is VERY difficult to encode it and deal with it, e.g. having things associate against k8s.pod and then realizing you need k8s.container modelled and half of the telemetry you generate should change its entity type (which can break downstream users interacting with entity).

Reply:
That's very useful examples, thank you.